build/chip/chip_test_suite.gni

```diff
@@ -84,6 +84,12 @@ template("chip_test_suite") {
     forward_variables_from(invoker, "*", [ "tests" ])
 
     output_dir = "${root_out_dir}/lib"
+
+    if (!defined(invoker.public_deps)) {
+      public_deps = []
+    }
+
+    public_deps += [ "${chip_root}/src/platform/logging:stdio" ]
```

Comment:
This is highly surprising, why? Can you at least at a TODO to move this dependency out of the test suite template?

Reply:
Without it I get 

```
...
.../lib/libnlunit-test.a  -ldl -lpthread -lrt -Wl,--end-group -o linux_x64_clang/tests/TestInetEndPoint
ld.lld: error: undefined symbol: chip::Logging::Platform::LogV(char const*, unsigned char, char const*, __va_list_tag*)
>>> referenced by CHIPLogging.cpp:137 (../../src/lib/support/logging/CHIPLogging.cpp:137)
>>>               libSupportLayer.CHIPLogging.cpp.o:(chip::Logging::LogV(unsigned char, unsigned char, char const*, __va_list_tag*)) in archive linux_x64_clang/obj/src/lib/support/lib/libSupportLayer.a
```

I believe the unit test for system may have a customized event loop (not pulling in the device layer) but are lacking a vlog implementation. Previous versions seem to have some hardcoded __attribute__((weak)) vlog implementations, but that seemed really hacky so I killed them.

A future pass should add the logging binding only for the tests that need it. Then again, do any tests without logging make sense? I think we need to follow up and figure if there is a split between tests without a platform layer vs tests with one.

Reply:
What you're seeing here is that the way you've avoided the cycle is to remove a dependency that's needed to link (It's not needed to compile.)

Re-adding the dependency at the test suite level fixes the link error, but it is not the right solution from my perspective - target should list all of their link-time dependencies generally speaking. This is what I meant by resolving cycles in a way that allows unit tests to link without additional dependencies - here, stdio is such an "additional dependency".